support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputAction.java

```diff
@@ -27,9 +33,51 @@ public void onAttached(Run<?, ?> r) {
     @Override
     public void onLoad(Run<?, ?> r) {
         this.run = r;
-        assert executions != null && !executions.contains(null) : executions;
-        for (InputStepExecution step : executions) {
-            step.run = run;
+        synchronized (this) {
+            if (ids == null) {
+                // Loading from before JENKINS-25889 fix. Load the IDs and discard the executions, which lack state anyway.
+                assert executions != null && !executions.contains(null) : executions;
```

Comment:
Not when we are deserializing an `InputAction` created before this patch, when the field was non-`transient`.